oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/AlarmCore.java

```diff
@@ -82,7 +82,9 @@ public void start(List<AlarmCallback> allCallbacks) {
                     }
                     List<AlarmMessage> filteredMessages = alarmMessageList.stream().filter(msg -> !msg.isOnlyAsCondition()).collect(Collectors.toList());
                     if (!filteredMessages.isEmpty()) {
-                        allCallbacks.forEach(callback -> callback.doAlarm(filteredMessages));
+                        for (AlarmCallback callback : allCallbacks) {
+                            callback.doAlarm(filteredMessages);
+                        }
```

Comment:
Why change this?

Reply:
> Why change this?

To propagate the exception up to the caller, we don't want to catch the exception in every plugin, because there is a place to catch all and print the logs in the alarm core

https://github.com/apache/skywalking/blob/da9fc57804a130d884c8ce0997073f9b32809c71/oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/AlarmCore.java#L80-L93